pulse2percept/stimuli.py

```diff
@@ -0,0 +1,531 @@
+        rf = receptive_field(e, xg, yg, rftype, rfsize)
+        magn.append(np.sum(rf.T * img_resize) / np.sum(rf))
+    magn = np.array(magn)
+
+    if max_contrast:
+        # Normalize contrast to valrange
+        if magn.min() < magn.max():
+            magn = (magn - magn.min()) / (magn.max() - magn.min())
+
+    # With `magn` between 0 and 1, now scale to valrange
+    magn = magn * np.diff(valrange) + valrange[0]
+
+    # Map magnitude to either freq or amp of pulse train
+    pulses = []
+    for m in magn:
+        if coding == 'amplitude':
+            # Map magnitude to amplitude
+            amp = m
+            freq = const_freq
+        elif coding == 'frequency':
+            # Map magnitude to frequency
+            amp = const_amp
+            freq = m
+        else:
+            e_s = "Acceptable values for `coding` are 'amplitude' or"
+            e_s += "'frequency'."
+            raise ValueError(e_s)
+
+        pt = Psycho2Pulsetrain(tsample, freq=freq, amp=amp, dur=dur,
+                               pulse_dur=pulsedur,
+                               interphase_dur=interphasedur,
+                               pulsetype=pulsetype)
+        pulses.append(pt)
+
+    return pulses
+
+
+@utils.deprecated
+class Movie2Pulsetrain(utils.TimeSeries):
+    """
+    Is used to create pulse-train stimulus based on luminance over time from
+    a movie
+
+    This class is deprecated as of v0.2 and will be replaced with a new
+    version in v0.3.
+    """
+
+    def __init__(self, rflum, tsample, fps=30.0, amp_transform='linear',
+                 amp_max=60, freq=20, pulse_dur=.5 / 1000.,
+                 interphase_dur=.5 / 1000.,
+                 pulsetype='cathodicfirst', stimtype='pulsetrain'):
+        """
+        Parameters
+        ----------
+        rflum : 1D array
+           Values between 0 and 1
+        tsample : suggest TemporalModel.tsample
+        """
+        if tsample <= 0:
+            raise ValueError("tsample must be a non-negative float.")
+
+        # set up the individual pulses
+        pulse = biphasic_pulse(pulsetype, pulse_dur, tsample,
+                               interphase_dur)
+        # set up the sequence
+        dur = rflum.shape[-1] / fps
+        if stimtype == 'pulsetrain':
+            interpulsegap = np.zeros(int(round((1.0 / freq) / tsample)) -
+                                     len(pulse.data))
+            ppt = []
+            for j in range(0, int(np.ceil(dur * freq))):
+                ppt = np.concatenate((ppt, interpulsegap), axis=0)
+                ppt = np.concatenate((ppt, pulse.data), axis=0)
+
+        ppt = ppt[0:int(round(dur / tsample))]
+        intfunc = spi.interp1d(np.linspace(0, len(rflum), len(rflum)),
+                               rflum)
+
+        amp = intfunc(np.linspace(0, len(rflum), len(ppt)))
+        data = amp * ppt * amp_max
+        utils.TimeSeries.__init__(self, tsample, data)
+
+
+class Psycho2Pulsetrain(utils.TimeSeries):
```

Comment:
Rename this to `PulseTrain` and keep pretty much as is.

Rename `mono` and `biphasic` pulses to `Mono` and `BiphasicPulse` and make them classes, instead of functions. 

In other words : separate what should be classes and what should be functions (`movie2pulsetrain` and `image2pulstrain` can remain functions that generate `PulseTrain` objects).